src/platform/react-hooks/src/AblyReactHooks.ts

```diff
@@ -16,3 +16,15 @@ export type ChannelNameAndId = {
   id?: string;
 };
 export type ChannelParameters = string | ChannelNameAndOptions;
+
+export const version = '1.2.45';
+
+export function channelOptionsWithAgent(options?: Types.ChannelOptions) {
+  return {
+    ...options,
+    params: {
```

Comment:
I think shallowEquals that we are using in `_shouldReattachToSetOptions` won't work this way :( because params will be new for each `useChannel` hook :( Could you add test that:

```
useChannel('my-channel');
useChannel('my-channel');
```
will be rendered without exception?

Reply:
I'm pretty sure it will work? It's a shallow equal comparison so even if the object reference is the same it will still return true if they have the same properties. You can test it manually with:
```js
console.log(
  Utils.shallowEquals(
    {
      agent: 'ably-js/1.2.3',
    },
    {
      agent: 'ably-js/1.2.3',
    }
  )
);
```

Adding tests for this won't really help with our current testing setup since the react-hooks test use a mock version of the ably-js SDK 😕

Reply:
oh, sorry, you are right! for some reason I thought we compare whole options, not params